### PR TITLE
Enhacements.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Add page error custom for bulma.
- [x] Add  visual indicator to labels of required fields.
- [x] Add tests page error.

### CODE REVIEW

File: resources/views/site/404.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+use Yiisoft\Html\Html;
+
+$this->setTitle('404');
+
+?>
+
+<h1 class='is-size-1'>
```
Comment (samdark) [inline resources/views/site/404.php:9]:
```suggestion
<h1 class="is-size-1">
```

File: resources/views/site/404.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <b>404</b>
+</h1>
+
+<p class='has-text-danger'>
```
Comment (samdark) [inline resources/views/site/404.php:13]:
```suggestion
<p class="has-text-danger">
```

File: resources/views/site/404.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    The page
+
+    <span>
+        <b><?= Html::encode($urlMatcher->getCurrentUri()->getPath()) ?></b>
+    </span>
```
Comment (samdark) [inline resources/views/site/404.php:18]:
```suggestion
    <strong><?= Html::encode($urlMatcher->getCurrentUri()->getPath()) ?></strong>
```

File: resources/views/site/404.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    not found. <br/>
```
Comment (samdark) [inline resources/views/site/404.php:20]:
```suggestion
    not found.
```

File: resources/views/site/404.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+</p>
+
+<p class='has-text-grey'>
+    <br/>The above error occurred while the Web server was processing your request. <br/>
+
+    Please contact us if you think this is a server error. Thank you. <br/>
+
+</p>
```
Comment (samdark) [inline resources/views/site/404.php:29]:
```suggestion
<p class="has-text-grey">
    The above error occurred while the Web server was processing your request.<br/>
    Please contact us if you think this is a server error. Thank you.
</p>
```

File: resources/views/site/404.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+<hr class='mb-2'>
```
Comment (samdark) [inline resources/views/site/404.php:31]:
```suggestion
<hr class="mb-2">
```

File: resources/views/site/404.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+<a class ='button is-danger mt-5' href=<?= $url->generate('site/index') ?>>
```
Comment (samdark) [inline resources/views/site/404.php:33]:
```suggestion
<a class ="button is-danger mt-5" href="<?= $url->generate('site/index') ?>">
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: resources/views/site/404.php (hunk 7 above)
Comment (terabytesoftw) [reply resources/views/site/404.php:33]: done.

File: resources/views/site/404.php (hunk 6 above)
Comment (terabytesoftw) [reply resources/views/site/404.php:31]: done.

File: resources/views/site/404.php (hunk 5 above)
Comment (terabytesoftw) [reply resources/views/site/404.php:29]: done.

File: resources/views/site/404.php (hunk 4 above)
Comment (terabytesoftw) [reply resources/views/site/404.php:20]: done.

File: resources/views/site/404.php (hunk 3 above)
Comment (terabytesoftw) [reply resources/views/site/404.php:18]: done.

File: resources/views/site/404.php (hunk 2 above)
Comment (terabytesoftw) [reply resources/views/site/404.php:13]: done.

File: resources/views/site/404.php (hunk 1 above)
Comment (terabytesoftw) [reply resources/views/site/404.php:9]: done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
- Add custom error page.
- Add visual indicator to labels of required fields.
- Add tests for error page.

**Comment by samdark** [comment]
👍

**Pull request merged**